### PR TITLE
feat: register

### PR DESCRIPTION
**Pull request by haidyhesham95**
feat: implement signup UI and integrate with API
+added register ui 
WhatsApp Image 2024-11-17 at 23 38 55_9e7b6eca

### CODE REVIEW

File: lib/core/networking/api_execute.dart
Comment (Marco-Nagy) [inline lib/core/networking/api_execute.dart:1]: this class name changed to Data class 

i suggests update project before pr

File: lib/core/networking/common/api_result.dart
Comment (Marco-Nagy) [inline lib/core/networking/common/api_result.dart:1]: this class name changed to Data class

i suggests update project before pr

File: lib/features/auth/presentation/signup/view_model/signup_cubit.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import 'package:injectable/injectable.dart';
+import 'package:meta/meta.dart';
+import '../../../../../core/networking/common/api_result.dart';
+import '../../../domain/entities/signup_entity.dart';
+
+part 'signup_state.dart';
+@injectable
+class SignupCubit extends Cubit<SignupState> {
+  SignupCubit(this._authUseCase) : super(SignupInitial());
+  final AuthUseCase _authUseCase;
```
Comment (Marco-Nagy) [inline lib/features/auth/presentation/signup/view_model/signup_cubit.dart:16]: its not best practice to declare  TextEditingController

Comment (Marco-Nagy) [reply lib/features/auth/presentation/signup/view_model/signup_cubit.dart:16]: i suggest refactoring this using Flutter hooks package 
i preferer to vote on this step or refactor it with best practice for declaration

File: lib/features/auth/presentation/signup/view_model/signup_state.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
+part of 'signup_cubit.dart';
+
+@immutable
+sealed class SignupState {}
+
+sealed class SignupAction {}
```
Comment (Marco-Nagy) [inline lib/features/auth/presentation/signup/view_model/signup_state.dart]: please separate Sign Up states and sign up action each one in dart file

**Review by Marco-Nagy — CHANGES_REQUESTED** [review]

**Review by somaya-mahmoud — APPROVED** [review]

**Review by YasmeenGad — APPROVED** [review]

**Review by Marco-Nagy — APPROVED** [review]

**Pull request merged**